be/src/vec/functions/array/function_array_element.h

```diff
@@ -0,0 +1,247 @@
+
+namespace doris::vectorized {
+
+class FunctionArrayElement : public IFunction {
+public:
+    static constexpr auto name = "element_at";
+    static FunctionPtr create() { return std::make_shared<FunctionArrayElement>(); }
+
+    /// Get function name.
+    String get_name() const override { return name; }
+
+    bool is_variadic() const override { return false; }
+
+    size_t get_number_of_arguments() const override { return 2; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        DCHECK(is_array(arguments[0]))
+                << "first argument for function: " << name << " should be DataTypeArray";
+        DCHECK(is_integer(arguments[1]))
+                << "second argument for function: " << name << " should be Integer";
+        return make_nullable(
+                check_and_get_data_type<DataTypeArray>(arguments[0].get())->get_nested_type());
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        auto dst_null_column = ColumnUInt8::create(input_rows_count);
+        UInt8* dst_null_map = dst_null_column->get_data().data();
+        const UInt8* src_null_map = nullptr;
+        ColumnsWithTypeAndName args;
+        auto col_left = block.get_by_position(arguments[0]);
+        if (col_left.column->is_nullable()) {
+            auto null_col = check_and_get_column<ColumnNullable>(*col_left.column);
+            src_null_map = null_col->get_null_map_column().get_data().data();
+            args = {{null_col->get_nested_column_ptr(), remove_nullable(col_left.type),
+                     col_left.name},
+                    block.get_by_position(arguments[1])};
+        } else {
+            args = {col_left, block.get_by_position(arguments[1])};
+        }
+
+        auto result_type = remove_nullable(
+                check_and_get_data_type<DataTypeArray>(args[0].type.get())->get_nested_type());
+
+        auto res_column = _perform(args, result_type, input_rows_count, src_null_map, dst_null_map);
+        if (!res_column) {
+            return Status::RuntimeError(
+                    fmt::format("unsupported types for function {}({}, {})", get_name(),
+                                block.get_by_position(arguments[0]).type->get_name(),
+                                block.get_by_position(arguments[1]).type->get_name()));
+        }
+        block.replace_by_position(
+                result, ColumnNullable::create(std::move(res_column), std::move(dst_null_column)));
+        return Status::OK();
+    }
+
+private:
+    ColumnPtr _perform(const ColumnsWithTypeAndName& arguments, const DataTypePtr& result_type,
+                       size_t input_rows_count, const UInt8* src_null_map, UInt8* dst_null_map) {
+        ColumnPtr res;
+        if (!((res = _execute_number<Int8>(arguments, result_type, input_rows_count, src_null_map,
```

Comment:
`_execute_number ` and `_execute_string` start with the same logic so could we move the common logics here. It will help us improve efficience. Now, for Float64 type, it have to be passed to each `_execute_number` call and return nullptr until it is passed to `_execute_number<Float64>`

Reply:
you are right, I have refractor these codes, please review again, thanks.